src/main/scala/com/cognite/sdk/scala/v1/Client.scala

```diff
@@ -310,12 +310,24 @@ object GenericClient {
     asJsonEither[CdpApiError, T].mapWithMetadata((response, metadata) =>
       response match {
         case Left(DeserializationException(_, error)) =>
-          throw SdkException(
-            s"Failed to parse response, reason: ${error.getMessage}",
-            Some(uri),
-            metadata.header("x-request-id"),
-            Some(metadata.code.code)
-          )
+          if (metadata.code.code === 429) {
+            throw CdpApiException(
+              url = uri"$uri",
+              code = 429,
```

Comment:
Better to use `metadata.code.code` here, in my opinion, or `StatusCode.TooManyRequests.code`, so the reader won't have to look up the meaning of 429.

Reply:
sure